test/spec/features/label-editing/TouchIntegrationSpec.js

```diff
@@ -15,21 +15,30 @@ describe('direct editing - touch integration', function() {
   function createModeler(xml, done) {
     var modeler = new Modeler({ container: container });
 
-    modeler.importXML(xml, function(err) {
-      done(err, modeler);
+    return new Promise(function(resolve) {
+
+      modeler.importXML(xml).then(function(result) {
+        resolve({ error: null, modeler: modeler });
+      }).catch(function(err) {
+        resolve({ error: err, modeler: modeler });
+      });
     });
   }
 
 
   it('should work on modeler (manual test)', function(done) {
     var xml = require('../../../fixtures/bpmn/simple.bpmn');
-    createModeler(xml, done);
+    createModeler(xml).then(function(result) {
+      done(result.error, result.warnings);
+    });
   });
 
 
   it('should edit labels via double tap (manual test)', function(done) {
     var xml = require('./LabelEditing.bpmn');
-    createModeler(xml, done);
+    createModeler(xml).then(function(result) {
+      done(result.error, result.warnings);
```

Comment:
What is the semantics of passing `warnings` as a second argument?

Better from my point of view: Get rid of `done`.